mipengine/common/node_registry_DTOs.py

```diff
@@ -0,0 +1,28 @@
+from pydantic import BaseModel
+from ipaddress import IPv4Address
+from typing import Optional, List
+from enum import Enum, unique
+
+
+@unique
+class NodeRole(Enum):
```

Comment:
The human readable value is already there GLOBALNODE, LOCALNODE. I would prefer not to assign values at all, but it seems it's not allowed with python enum. Anyhow if you want to get the name as string, which I want to discouragebecause I want to discourage string comparisons you can do
`NodeRole.GLOBALNODE.name`
this returns a string "GLOBALNODE"

the auto() just increments the value starting from 1

Reply:
On the `node_registry` line 89 is this code:
```
      role = ""
      if NodeRole.GLOBALNODE.name in tags:
          role = NodeRole.GLOBALNODE
      elif NodeRole.LOCALNODE.name in tags:
          role = NodeRole.LOCALNODE
      else:
          continue
```

This is a string comparison right? My point was that if you use the "string Enums" you will be able to do:
```
      role = ""
      if NodeRole.GLOBALNODE in tags:
          role = NodeRole.GLOBALNODE
      elif NodeRole.LOCALNODE in tags:
          role = NodeRole.LOCALNODE
      else:
          continue
```

Also on the `node.py`:
```
node_role = (
      NodeRole.LOCALNODE
      if node_config.role == NodeRole.LOCALNODE.name
      else NodeRole.GLOBALNODE
)
```
This is a string comparison again, right?

Also, with pydantic and string enums you could do something like this: https://pydantic-docs.helpmanual.io/usage/types/#enums-and-choices

You do not compare, just assign a value and pydantic automatically checks if it's of the allowed Enum values. You can only do this though if the Enum has string name/value instead of integers.

Reply:
done